File: problems/0209.长度最小的子数组/0209-长度最小的子数组.cpp
class Solution {
public:
    int minSubArrayLen(int s, vector<int>& nums) {
        int n = nums.size();
        int ans = INT_MAX;
        int sum = 0;

        for(int start = 0, end = 0; end<n; end++){
            sum += nums[end];
            while (sum >= s) {
                ans = min(ans, end - start + 1);
                sum -= nums[start];
                start++;
            }
        }
        return ans == INT_MAX ? 0 : ans;
    }
};

File: 0209.长度最小的子数组/0209-长度最小的子数组.js
/**
 * @param {number} target
 * @param {number[]} nums
 * @return {number}
 */
var minSubArrayLen = function(target, nums) {
    const n=nums.length;
    let ans=Infinity, _sum=0;
    for(let l=0, r=0; r<n; r++){
        _sum+=nums[r];
        while(_sum>target){
            if(_sum-nums[l]<target) break;
            _sum-=nums[l];
            l++;
        }
        // console.log(_sum, l, r);
        if(_sum>=target) ans=Math.min(ans, r-l+1);
    }
    return ans===Infinity?0:ans;
};